### PR TITLE
[BugFix] Fix Hudi MOR timestamp type

### PR DESCRIPTION
**Pull request by dirtysalt**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

---
Don't change original hive type when passing `column.types`.

If so `TimestampMicros` type will be changed to `timestamp` type when setting `column.types`. which is incorrect.  Because in `ColumnType.java`, two hive type will be mapped into different types.
- TimestampMicros -> DATETIME_MICROS
- timestamp -> DATETIME

---
DATETIME type might have `LongWritable` underneath. And time unit is microseconds.

However this test case can not cover it(I can not fabricate the test case on my spark-sql). I find this bug on user case. 

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

File: java-extensions/hudi-reader/src/main/java/com/starrocks/hudi/reader/HudiScannerUtils.java [hunk 1]
```diff
@@ -58,6 +60,11 @@ public static LocalDateTime getTimestamp(long value, TimeUnit timeUnit, boolean
         long nanoseconds = 0L;
 
         switch (timeUnit) {
+            case SECONDS:
+                seconds = value;
+                nanoseconds = 0;
+                break;
+
```
Comment (miomiocat) [inline java-extensions/hudi-reader/src/main/java/com/starrocks/hudi/reader/HudiScannerUtils.java:67]: where to use `SECONDS`?

Comment (dirtysalt) [reply java-extensions/hudi-reader/src/main/java/com/starrocks/hudi/reader/HudiScannerUtils.java:67]: actually not used any. When I tested it, I've used that.

**Comment by dirtysalt** [comment]
run starrocks_admit_test

**Review by satanson — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Comment by dirtysalt** [comment]
@MergifyIO backport branch-2.5

**Pull request merged**